R/presto.field_utilities.R

```diff
@@ -0,0 +1,166 @@
+print.presto.field <- function(prf, prefix = "", ...) {
+  cat(prefix, "presto.field: \n", sep = "")
+  cat(prefix, "..Name: ", prf$name_, "\n", sep = "")
+  cat(prefix, "..Type:  ", prf$type_, "\n", sep = "")
+  if (prf$is_map_) {
+    cat(prefix, "..Key Type:  ", prf$key_type_, "\n", sep = "")
+  }
+  cat(prefix, "..Is Array:  ", prf$is_array_, "\n", sep = "")
+  cat(prefix, "..Is Map:  ", prf$is_map_, "\n", sep = "")
+  cat(prefix, "..Is Row:  ", prf$is_row_, "\n", sep = "")
+  cat(prefix, "..Is parent Map:  ", prf$is_parent_map_, "\n", sep = "")
+  cat(prefix, "..Is parent Array:  ", prf$is_parent_array_, "\n", sep = "")
+  cat(prefix, "..Fields:  ", length(prf$fields_), " fields\n", sep = "")
+  if (!is.na(prf$timezone_)) {
+    cat(prefix, "..Timezone:  ", prf$timezone_, "\n", sep = "")
+  }
+  if (length(prf$fields_) > 0) {
+    for (f in prf$fields_) {
+      print(f, prefix = paste0(prefix, "...."))
+    }
+  }
+  invisible(prf)
+}
+
+is_simple_type <- function(prf) {
+  return((!prf$type_ %in% c("PRESTO_ROW", "PRESTO_MAP")))
+}
+
+parse.timestamp <- function(x, timezone) {
+  if (!requireNamespace("lubridate", quietly = TRUE)) {
+    stop("The [", x$name_, "] field is a TIMESTAMP WITH TIMEZONE ",
+         "field. Please install the lubridate package before ",
+         "importing it to R.",
+         call. = FALSE)
+  }
+  if (is.na(x)) {
+    return(as.POSIXct(x, tz = timezone))
+  }
+  parts <- unlist(strsplit(x, " "))
```

Comment:
can you add a comment on intent here (describing what `parts` are)?

Reply:
Good point. Added.